cli/command/utils.go

```diff
@@ -90,9 +95,35 @@ func PromptForConfirmation(ins io.Reader, outs io.Writer, message string) bool {
 		ins = streams.NewIn(os.Stdin)
 	}
 
-	reader := bufio.NewReader(ins)
-	answer, _, _ := reader.ReadLine()
-	return strings.ToLower(string(answer)) == "y"
+	result := make(chan bool, 1)
+	// We use a context to cancel the prompt or the signal handler
+	ctx, cancel := context.WithCancel(context.TODO())
+
+	go func(_ context.Context) {
+		// cleanup the other goroutine
+		defer cancel()
+		reader := bufio.NewReader(ins)
+		answer, _, _ := reader.ReadLine()
+		result <- (strings.ToLower(string(answer)) == "y")
+	}(ctx)
```

Comment:
Just did a test using the https://github.com/uber-go/goleak package on a toy application.

Found that it does not leak at all, presumable since the `stdin` should be closed by the time the program exits. 

```go
func TestLeakyReader(t *testing.T) {
	defer goleak.VerifyNone(t)

	result := make(chan bool)
	ctx, cancel := context.WithCancel(context.TODO())

	reader := bufio.NewReader(os.Stdin)
	go func() {
		// cleanup the other goroutine
		defer cancel()
		answer, _, _ := reader.ReadLine()
		result <- (strings.ToLower(string(answer)) == "y")
	}()

	go func() {
		// cleanup the other goroutine
		defer cancel()
		// We catch the signal to avoid the prompt
		// from exiting as an error
		c := make(chan os.Signal, 1)
		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
		select {
		case <-ctx.Done():
			return
		case <-c:
			result <- false
		}
	}()

	fmt.Printf("%t\n", <-result)
}
```

Adding another goroutine which sleeps causes there to be a leak (correctly so).
```go
go func() {
		time.Sleep(time.Minute * 1)
	}()
```

```shell
|| --- FAIL: TestLeakyReader (0.45s)
reader_test.go|51| found unexpected goroutines:
||         [Goroutine 20 in state sleep, with time.Sleep on top of the stack:
```

Reply:
I assume this would however be a problem if the program does not exit after a prompt. 

For example, prompting a user to confirm, returning false, continue execution and then prompting the user again to confirm.